builtin/providers/aws/resource_aws_dc_intra_virtual_interface.go

```diff
@@ -0,0 +1,284 @@
+			NewPrivateVirtualInterfaceAllocation: &directconnect.NewPrivateVirtualInterfaceAllocation{
+				Asn:                  aws.Int64(int64(d.Get("asn").(int))),
+				VirtualInterfaceName: aws.String(d.Get("virtual_interface_name").(string)),
+				Vlan:                 aws.Int64(int64(d.Get("vlan").(int))),
+			},
+			OwnerAccount: aws.String(d.Get("owner_account_id").(string)),
+		}
+
+		if v, ok := d.GetOk("amazon_address"); ok {
+			createOpts.NewPrivateVirtualInterfaceAllocation.AmazonAddress = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("auth_key"); ok {
+			createOpts.NewPrivateVirtualInterfaceAllocation.AuthKey = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("customer_address"); ok {
+			createOpts.NewPrivateVirtualInterfaceAllocation.CustomerAddress = aws.String(v.(string))
+		}
+
+		log.Printf("[DEBUG] Creating DirectConnect private virtual interface")
+		resp, err = conn.AllocatePrivateVirtualInterface(createOpts)
+		if err != nil {
+			return fmt.Errorf("Error creating DirectConnect Virtual Interface: %s", err)
+		}
+
+	}
+
+	// Store the ID
+	VirtualInterface := resp
+	d.SetId(*VirtualInterface.VirtualInterfaceId)
+	log.Printf("[INFO] VirtualInterface ID: %s", *VirtualInterface.VirtualInterfaceId)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"pending"},
+		Target:     []string{"available", "confirming", "verifying", "pending"},
+		Refresh:    DirectConnectIntraVirtualInterfaceRefreshFunc(conn, *VirtualInterface.VirtualInterfaceId),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+
+	_, stateErr := stateConf.WaitForState()
+	if stateErr != nil {
+		return fmt.Errorf(
+			"Error waiting for DirectConnect PrivateVirtualInterface (%s) to become ready: %s",
+			*VirtualInterface.VirtualInterfaceId, err)
+	}
+
+	// Read off the API to populate our RO fields.
+	return resourceAwsDirectConnectIntraVirtualInterfaceRead(d, meta)
+}
+
+func DirectConnectIntraVirtualInterfaceRefreshFunc(conn *directconnect.DirectConnect, virtualinterfaceId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+
+		resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
+			VirtualInterfaceId: aws.String(virtualinterfaceId),
+		})
+
+		if err != nil {
+
+			log.Printf("Error on DirectConnectPrivateVirtualInterfaceRefresh: %s", err)
+			return nil, "", err
+
+		}
+
+		if resp == nil || len(resp.VirtualInterfaces) == 0 {
+			return nil, "", nil
+		}
+
+		virtualInterface := resp.VirtualInterfaces[0]
+		return virtualInterface, *virtualInterface.VirtualInterfaceState, nil
+	}
+}
+
+func resourceAwsDirectConnectIntraVirtualInterfaceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dcconn
+
+	resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
+		VirtualInterfaceId: aws.String(d.Id()),
+	})
+
+	if err != nil {
+
```

Comment:
Is there a chance that someone can delete the VirtualInterface from the AWS Console? If so, this will error out. We should handle the case of 404 and then remove from state if necessary